### PR TITLE
Syndicate Cyborg: The Rebaking

### PR DESCRIPTION
**Pull request by jknpj**
motile droid-crisis

Heavily unatomic, like all good silicon PR. **This PR isn't re-adding the Syndicate Cyborg to Nuke Ops, yet.** I'll make a proper changelog when i'm done. Both syndieborgs have WIP loadouts so don't mind those commented gatling gun and eimplanter modules. Changes made so far:

* Moved all robot modules in both robot_item.dm files to their own files.
* Ported and tweaked generic robot racks from bay.
* Refactors mediborg's roller bed holder as a robot rack.
* Salvaged the robot antag spawner and refactored it into a generic robot spawner.
* Refactors hugborg's strange ball/egg as robot spawners.
* Removed all access from cyborgs.
* Cyborg modules now hold their access and by defaullt, NT cyborgs have 'all station access'.
* Module defines.
* Renamed the old **Syndicate Module** as the **Syndicate Blitzkrieg Module**.
* Created a new syndicate module, the **Syndicate Crisis Module**.
* Added a new module quirk: MODULE_IS_DEFINITIVE
* Cyborg modules with MODULE_IS_DEFINITIVE cannot be reseted.
* Added MODULE_IS_DEFINITIVE to hugborgs and both syndicate modules.
* Made syndicate cyborgs need syndicate access to unlock their cover.
* Changed the starting sound of both NT combat cyborgs and syndicate cyborgs
* Made the syndicate subtype have its own robot holochip so they can work with OPs(And they can know when their burgers are dead)
* Added a way for items to get refunded in the syndicate uplink

:cl: 
* rscdel: NT cyborgs now only have all station access, being unable to tamper with syndicate/centcomm equipment without an AI's help.
* rscdel: Hugborgs aren't compatible with the "cyborg module reset board" anymore.

### CODE REVIEW

**Comment by SonixApache** [comment]
>Changed the starting sound of both NT combat cyborgs and syndicate cyborgs

Nonononononononono.
Those are MECHA lines. For MECHA. Get the fuck outta here.

**Comment by mtpowell** [comment]
Ah the silicon meta-club is still alive

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1767,3 +1767,19 @@ Game Mode config tags:
 			continue
 		turfs += T
 	return pick(turfs)
+
+/proc/get_closest_atom(type, list, source)
```
Comment (jknpj) [inline code/__HELPERS/unsorted.dm]: I ended not using this, not sure if i leave it or nah 🤔

File: vgstation13.dme [hunk 2]
```diff
@@ -2307,7 +2328,7 @@
 #include "interface\web\interface.dms"
 #include "maprendering\maprendering.dm"
 #include "maps\_map.dm"
-#include "maps\tgstation.dm"
+#include "maps\test_tiny.dm"
```
Comment (jknpj) [inline vgstation13.dme]: every time

File: code/__HELPERS/unsorted.dm (hunk 1 above)
Comment (DamianX) [reply code/__HELPERS/unsorted.dm]: delet

**Comment by Hoshi-chan** [comment]
I'm super excited for this.

File: code/game/objects/items/robot/robot_items/robot_rack.dm [hunk 3]
```diff
@@ -68,7 +68,11 @@
 	. = ..()
 
 /obj/item/robot_rack/bed/update_icon()
-	icon_state = "borgbed_[length(held) > 0 ? "stored" : "deployed"]"
+	icon_state = "[initial(icon_state)][length(held) > 0 ? "stored" : "deployed"]"
```
Comment (DamianX) [inline code/game/objects/items/robot/robot_items/robot_rack.dm]: wouldn't this make it invisible before `update_icon` is called? or is `update_icon` already called somewhere upon creation?

Comment (jknpj) [reply code/game/objects/items/robot/robot_items/robot_rack.dm]: It is called when the rack spawns the initial icon

Comment (DamianX) [reply code/game/objects/items/robot/robot_items/robot_rack.dm]: 👌

**Comment by jknpj** [comment]
Removing the balance tag, since they're going to stay as adminbus for now.

**Comment by jknpj** [comment]
Actually adding it back because i messed with silicon access, after all.

**Comment by bathosbathtime** [comment]
Why does this remove access from moduleless cyborgs

**Pull request merged**